security-analysis/security-analysis-api/src/main/java/com/powsybl/security/LimitViolationDetection.java

```diff
@@ -274,4 +264,17 @@ static void checkVoltageAngle(VoltageAngleLimit voltageAngleLimit, double value,
                     }
                 });
     }
+
+    public static ViolationLocation createViolationLocation(Bus bus, VoltageLevel vl) {
```

Comment:
There's no need to have the `VoltageLevel` as a parameter.
Since the method is public static, it could be used by anyone. It would be best to limit potential usage errors.